app/backend-dotnet/Program.cs

```diff
@@ -0,0 +1,72 @@
+using Azure;
+using Azure.AI.OpenAI;
+using Azure.Core;
+using Azure.Identity;
+using Azure.Search.Documents;
+using Azure.Storage.Blobs;
+using IMJohn.Services;
+using Microsoft.Extensions.FileProviders;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Configuration;
+
+var builder = WebApplication.CreateBuilder(args);
+
+// Add services to the container.
+
+builder.Services.AddControllers();
+// Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+builder.Services.AddEndpointsApiExplorer();
+builder.Services.AddSwaggerGen();
+
+var azureCredential = new DefaultAzureCredential();
+var AZURE_STORAGE_ACCOUNT = builder.Configuration.GetValue<string>("AZURE_STORAGE_ACCOUNT");
+
+// Add blob service client
+var blobServiceClient = new BlobServiceClient(new Uri($"https://{AZURE_STORAGE_ACCOUNT}.blob.core.windows.net"), azureCredential);
+builder.Services.Add(ServiceDescriptor.Singleton(blobServiceClient));
+
+// Add blob container client
+var AZURE_STORAGE_CONTAINER = builder.Configuration.GetValue<string>("AZURE_STORAGE_CONTAINER");
+var blobContainerClient = blobServiceClient.GetBlobContainerClient(AZURE_STORAGE_CONTAINER);
+builder.Services.Add(ServiceDescriptor.Singleton(blobContainerClient));
+
+// Add search client
+var AZURE_SEARCH_SERVICE = builder.Configuration.GetValue<string>("AZURE_SEARCH_SERVICE");
+var AZURE_SEARCH_INDEX = builder.Configuration.GetValue<string>("AZURE_SEARCH_INDEX");
+var searchClient = new SearchClient(new Uri($"https://{AZURE_SEARCH_SERVICE}.search.windows.net"), AZURE_SEARCH_INDEX, azureCredential);
+builder.Services.Add(ServiceDescriptor.Singleton(searchClient));
+
+// add semantic kernel
+var AZURE_OPENAI_CHATGPT_DEPLOYMENT = builder.Configuration.GetValue<string>("AZURE_OPENAI_CHATGPT_DEPLOYMENT");
+var AZURE_OPENAI_GPT_DEPLOYMENT = builder.Configuration.GetValue<string>("AZURE_OPENAI_GPT_DEPLOYMENT");
+var AZURE_OPENAI_SERVICE = builder.Configuration.GetValue<string>("AZURE_OPENAI_SERVICE");
```

Comment:
```suggestion
var AZURE_OPENAI_CHATGPT_DEPLOYMENT = builder.Configuration["AZURE_OPENAI_CHATGPT_DEPLOYMENT"];
var AZURE_OPENAI_GPT_DEPLOYMENT = builder.Configuration["AZURE_OPENAI_GPT_DEPLOYMENT"];
var AZURE_OPENAI_SERVICE = builder.Configuration["AZURE_OPENAI_SERVICE"];
```

These all caps variable names aren't idiomatic in .NET. These locals should be camelcase.

Reply:
I prefer to use all caplital cases if a variable is a configuration and I can easily differentiable them from common variables.

Reply:
It’s not idiomatic in .NET to do this.